### DISCUSSION TITLE
SqsEventSource setting maximum concurrency for AWS Lambda

### DISCUSSION THREAD

**Discussion by gbvanrenswoude** [aws/aws-cdk#23751] [Q&A] [answered]
Hey all, I have an SqsEventSource (https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_event_sources.SqsEventSource.html) and I want to set maximumConcurrency on it as announced here: https://aws.amazon.com/blogs/compute/introducing-maximum-concurrency-of-aws-lambda-functions-when-using-amazon-sqs-as-an-event-source/ . However, this is not a property yet in cdk 2.61.0. I also find no .node.defaultChild on this Construct. How do I go about implementing this? There is AWS SAM support for it, as shown here: https://github.com/aws-samples/aws-lambda-amazon-sqs-max-concurrency/blob/main/template.yaml#LL50-L51C36

**Comment by gbvanrenswoude** [comment, marked as answer]
This will come in aws cdk 2.62.0.

**Comment by PlengNakdee** [comment]
I'm using aws-cdk 2.63.0, but still got error `'maxConcurrency' does not exist in type 'SqsEventSourceProps'`

**Reply by richard-baker-cardeahealth-io** [reply]
It worked!!
I had to start a brand new cdk instance, the existing project was still using an older version (even though `cdk --version` showed the latest version) has still using an older version of the cdk and sdk (in the package.jsons, etc..), even after running `npm install -g aws-cdk`

Does anyone have the upgrade steps to take to avoid this and keep it updated in the future?

**Reply by MSRSK** [reply]
What exactly do you mean by brand new instance?

**Reply by MSRSK** [reply]
I did npm upgrade and it worked
